x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -305,6 +329,52 @@ private void getVersionedDocument(String uid, ActionListener<VersionedDocument>
         });
     }
 
+    private void getVersionedDocuments(
+        Collection<String> uids,
+        ActionListener<Tuple<Map<String, VersionedDocument>, List<String>>> listener
+    ) {
+        if (uids.isEmpty()) {
+            listener.onResponse(new Tuple<>(Map.of(), List.of()));
+            return;
+        }
+        tryFreezeAndCheckIndex(listener).ifPresent(frozenProfileIndex -> {
+            new OriginSettingClient(client, SECURITY_PROFILE_ORIGIN).prepareMultiGet()
+                .addIds(frozenProfileIndex.aliasName(), uids.stream().map(ProfileService::uidToDocId).toArray(String[]::new))
+                .execute(ActionListener.wrap(multiGetResponse -> {
```

Comment:
I think we need wrap this inside a `frozenProfileIndex .checkIndexVersionThenExecute` call.

Reply:
Good catch!